browser/page_mapping.go

```diff
@@ -475,7 +477,10 @@ func mapPageOn(vu moduleVU, p *common.Page) func(common.PageOnEventName, sobek.C
 			})
 
 			if pageOnEvent.wait {
-				<-done
+				select {
+				case <-done:
+				case <-ctx.Done():
```

Comment:
TODO in new PR: Return an error so that we can abort the tagging of the metric with a name if one was set.

Reply:
https://github.com/grafana/xk6-browser/pull/1495